iceoryx2/src/sample_mut.rs

```diff
@@ -12,91 +12,152 @@
 
 //! # Example
 //!
+//! ## Sample Write Function
+//!
 //! ```
 //! use iceoryx2::prelude::*;
+//! use iceoryx2::sample_mut_impl::SampleMutImpl;
+//!
+//! fn write_sample_data<UninitSample: UninitializedSampleMut<u64>>(mut sample: UninitSample) -> impl SampleMut<u64> {
```

Comment:
Just as food for thought. This example is quite small and a simple `S` or `T` would be more concise without loosing information.